GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs

```diff
@@ -8,12 +8,49 @@
 
 namespace GitUI.CommandsDialogs.BrowseDialog.DashboardControl
 {
+    public interface IRepositoryFilter
+    {
+        void Filter(IList<Repository> repositories);
+    }
+
+    /// <summary>
+    /// Case insensitive text search based on the repository path
+    /// </summary>
+    public class RepositoryTextFilter : IRepositoryFilter
+    {
+        private readonly string _pattern;
+
+        public RepositoryTextFilter(string pattern)
```

Comment:
1. `pattern` is a variable, so it mustn't be set via the .ctor, and pass to `Filter` as an arg instead.
2. In general I'm not convinced this class is needed at all. This functionality can be part of `IUserRepositoriesListController`, e.g. `void IUserRepositoriesListController.FilterRepositories(filter, ListView.ListViewItemCollection repositories)`